### PR TITLE
LazilyInitializedEncoding overrides overloads with Endpoint

### PR DESCRIPTION
**Pull request by carterkozak**
Previously this would drop the Endpoint arg, and funnel calls through the original method.

==COMMIT_MSG==
LazilyInitializedEncoding overrides overloads with Endpoint
==COMMIT_MSG==

### CODE REVIEW

File: conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/LazilyInitializedEncoding.java [hunk 1]
```diff
@@ -91,5 +107,10 @@ private static final class LazilyInitializedDeserializer<T> implements Deseriali
         public T deserialize(InputStream input) throws IOException {
             return delegate.get().deserialize(input);
         }
+
+        @Override
+        public String toString() {
+            return "LazilyInitializedDeserializer{" + delegate + '}';
+        }
```
Comment (carterkozak) [inline conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/LazilyInitializedEncoding.java:114]: drive-by: added toString to these as best practice for debugging

**Review by mpritham — APPROVED** [review]
